precompiles/assets/types.go

```diff
@@ -32,27 +36,121 @@ func (p Precompile) GetDepositWithdrawParamsFromInputs(ctx sdk.Context, args []i
 	assetAddr, ok := args[1].([]byte)
 	if !ok || assetAddr == nil {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 1, "[]byte", assetAddr)
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 1, reflect.TypeOf(args[1]), args[1])
 	}
 	if len(assetAddr) != types.GeneralAssetsAddrLength {
-		return nil, xerrors.Errorf(exocmn.ErrInputClientChainAddrLength, len(assetAddr), types.GeneralClientChainAddrLength)
+		return nil, xerrors.Errorf(exocmn.ErrInvalidAddrLength, len(assetAddr), types.GeneralClientChainAddrLength)
 	}
 	depositWithdrawParams.AssetsAddress = assetAddr[:clientChainAddrLength]
 
 	stakerAddr, ok := args[2].([]byte)
 	if !ok || stakerAddr == nil {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 2, "[]byte", stakerAddr)
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 2, reflect.TypeOf(args[2]), args[2])
 	}
 	if len(stakerAddr) != types.GeneralClientChainAddrLength {
-		return nil, xerrors.Errorf(exocmn.ErrInputClientChainAddrLength, len(assetAddr), types.GeneralClientChainAddrLength)
+		return nil, xerrors.Errorf(exocmn.ErrInvalidAddrLength, len(assetAddr), types.GeneralClientChainAddrLength)
 	}
 	depositWithdrawParams.StakerAddress = stakerAddr[:clientChainAddrLength]
 
 	opAmount, ok := args[3].(*big.Int)
 	if !ok || opAmount == nil || opAmount.Cmp(big.NewInt(0)) == 0 {
-		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 3, "*big.Int", opAmount)
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 3, reflect.TypeOf(args[3]), args[3])
 	}
 	depositWithdrawParams.OpAmount = sdkmath.NewIntFromBigInt(opAmount)
 
 	return depositWithdrawParams, nil
 }
+
+func (p Precompile) ClientChainInfoFromInputs(_ sdk.Context, args []interface{}) (*types.ClientChainInfo, error) {
+	if len(args) != 5 {
+		return nil, xerrors.Errorf(cmn.ErrInvalidNumberOfArgs, 5, len(args))
+	}
+	clientChain := types.ClientChainInfo{}
+	clientChainID, ok := args[0].(uint32)
+	if !ok {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 0, reflect.TypeOf(args[0]), args[0])
+	}
+	clientChain.LayerZeroChainID = uint64(clientChainID)
+
+	addressLength, ok := args[1].(uint32)
+	if !ok || addressLength == 0 {
+		return nil, fmt.Errorf(exocmn.ErrContractInputParaOrType, 1, reflect.TypeOf(args[1]), args[1])
+	}
+	if addressLength > types.GeneralAssetsAddrLength {
+		return nil, xerrors.Errorf(exocmn.ErrInvalidAddrLength, addressLength, "not greater than 32")
+	}
+	clientChain.AddressLength = addressLength
+
+	name, ok := args[2].(string)
+	if !ok || name == "" {
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 2, reflect.TypeOf(args[2]), args[2])
+	}
+	clientChain.Name = name
+
+	metaInfo, ok := args[3].(string)
+	if !ok {
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 3, reflect.TypeOf(args[2]), args[2])
+	}
+	clientChain.MetaInfo = metaInfo
+
+	signatureType, ok := args[4].(string)
+	if !ok {
```

Comment:
Same here, should we validate non-empty?

Reply:
This field can be left empty, and there will be a default signature type when the input is empty. However, the final decision will be made after researching the signature algorithms of most blockchains.